app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java

```diff
@@ -91,27 +90,14 @@ public int getApiLevelActual() {
         return (Integer) API_LEVEL_ACTUAL;
     }
 
-    @Nullable
     private UiObject2 toUiObject2(Object selector, AccessibilityNodeInfo node) {
         Object[] constructorParams = {getUiDevice(), selector, node};
-        long end = SystemClock.uptimeMillis() + UIOBJECT2_CREATION_TIMEOUT;
-        while (true) {
```

Comment:
Don't need to wait for the element?

Reply:
nope, this constructor is never going to throw an exception. And retry does not help anyway even if it does